### PR TITLE
MM-45208: strip post meta for plugins

### PR DESCRIPTION
**Pull request by levb**
#### Summary
Fixed the race condition introduced in #20612

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

File: app/post.go [hunk 1]
```diff
@@ -309,21 +310,14 @@ func (a *App) CreatePost(c request.CTX, post *model.Post, channel *model.Channel
...
-
+	// We make a copy of the post for the plugin hook to avoid a race condition,
+	// and to remove the non-GOB-encodable Metadata from it.
 	if pluginsEnvironment := a.GetPluginsEnvironment(); pluginsEnvironment != nil {
+		pluginPost := rpost.ForPlugin()
 		a.Srv().Go(func() {
 			pluginContext := pluginContext(c)
 			pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
-				hooks.MessageHasBeenPosted(pluginContext, rPostCopy)
+				hooks.MessageHasBeenPosted(pluginContext, pluginPost)
```
Comment (ashishbhate) [inline app/post.go:320]: This was the fix for the race?

**Review by ashishbhate — APPROVED** [review]
LGTM

File: app/post.go [hunk 2]
```diff
@@ -673,7 +670,7 @@ func (a *App) UpdatePost(c *request.Context, post *model.Post, safeUpdate bool)
 		a.Srv().Go(func() {
 			pluginContext := pluginContext(c)
 			pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
-				hooks.MessageHasBeenUpdated(pluginContext, newPost, oldPost)
+				hooks.MessageHasBeenUpdated(pluginContext, newPost.ForPlugin(), oldPost.ForPlugin())
```
Comment (ashishbhate) [inline app/post.go:673]: Shouldn't we make similar changes here?

Comment (agnivade) [reply app/post.go:673]: Agree.

Comment (levb) [reply app/post.go]: @ashishbhate yes, missed it, thx! Fixed now.

**Comment by levb** [comment]
@DHaussermann FYI, no extra QA is relative to the original PR testing.

**Pull request merged**